node/lib/util/synthetic_branch_util.js

```diff
@@ -184,10 +212,15 @@ function* checkSubmodules(repo, commit) {
     assert.instanceOf(commit, NodeGit.Commit);
 
     const config = yield repo.config();
-    const whitelistPattern = (
+    const urlWhitelistPattern = (
         yield ConfigUtil.getConfigString(
             config, "gitmeta.skipsyntheticrefpattern")) || "";
-    const cfg = new SyntheticBranchConfig(whitelistPattern);
+    const pathWhitelistPattern = (
+        yield ConfigUtil.getConfigString(
+            config, "gitmeta.skipsyntheticrefpathpattern")) || "";
```

Comment:
isn't this a little dangerous? An empty string will match all strings in RegEx, aka `RegEx('').test(anything)` always return true.

This behavior is different from what defined in node/lib/util/synthetic_branch_util.js:73 where if regex test always return false. So probably we can do 

```
ield ConfigUtil.getConfigString(config, "gitmeta.skipsyntheticrefpathpattern")) || ConfigUtil.DEFAULT_PATTERN;
```
here and in ConfigUtil, we can check if patten === DEFAULTPATTERN